p2p/dht/bootstrap.go

```diff
@@ -44,38 +44,35 @@ func (d *KadDHT) Bootstrap(ctx context.Context) error {
 		return ErrZeroConnections
 	}
 	// register bootstrap nodes
-	bn := 0
+	bn := make([]discNode, 0, len(d.config.BootstrapNodes))
 	for _, n := range d.config.BootstrapNodes {
 		nd, err := node.NewNodeFromString(n)
 		if err != nil {
 			// TODO : handle errors
 			continue
 		}
-
-		d.rt.Update(discNode{nd, nd.Address()})
-		defer d.rt.Remove(discNode{nd, nd.Address()})
-
-		bn++
+		bn = append(bn, discNode{nd, nd.Address()})
 		d.local.Info("added new bootstrap node %v", nd)
 	}
 
-	if bn == 0 {
+	if len(bn) == 0 {
 		return ErrConnectToBootNode
 	}
 
-	d.local.Debug("netLookup using %d preloaded bootnodes ", bn)
+	d.local.Debug("lookup using %d preloaded bootnodes ", len(bn))
 
 	err := d.tryBoot(ctx, bn, c)
-
 	return err
 }
 
-func (d *KadDHT) tryBoot(ctx context.Context, bootnodecount int, minPeers int) error {
+func (d *KadDHT) tryBoot(ctx context.Context, bootnodes []discNode, minPeers int) error {
 
 	searchFor := d.local.PublicKey()
+	servers := make([]discNode, len(bootnodes))
+	copy(servers, bootnodes)
+
 	tries := 0
-	var size int
-	d.local.Debug("BOOTSTRAP: Running kademlia lookup for ourselves")
+	size := 0
 
```

Comment:
its not that trivial that there is only one running instance of the goroutine 
id add a comment or something to make it clearer
cause if they run concurrently there will be data races